src/OrthoTree/morton_codes.h

```diff
@@ -44,15 +46,18 @@ struct Morton {
      *
      * @param max_depth
      * @return Morton&
-     
+     *
+    */
+    // TODO discuss structure of singleton. 
+    // Maybe just write this file procedurally as helper functions? 
```

Comment:
singleton is 100% not a good idea
procedural could be an idea, how?
alternatively: static class or a wrapper for integer type (assuming both have zero overhead)

Reply:
yes. just give each instance of the orthoTree a MortonHelper or something like that which contains the functionality of this class and then inline everything